src/main/java/net/harawata/reflection/TypeParameterResolver.java

```diff
@@ -75,11 +75,15 @@ public static Type resolveReturnType(Method method, Type srcType) {
   public static Type[] resolveParamTypes(Method method, Type srcType) {
     Type[] paramTypes = method.getGenericParameterTypes();
     Class<?> declaringClass = method.getDeclaringClass();
-    Type[] result = new Type[paramTypes.length];
-    for (int i = 0; i < paramTypes.length; i++) {
-      result[i] = resolveType(paramTypes[i], srcType, declaringClass);
+    return resolveTypes(paramTypes, srcType, declaringClass);
+  }
+
+  private static Type[] resolveTypes(Type[] types, Type srcType, Class<?> declaringClass) {
```

Comment:
-30 lines. Nice one! 👍 
I'm terribly sorry for being OCD, but could you move `resolveTypes()` before or after `resolveType()`? 🙏😅